### ISSUE TITLE
Plugin crash

### ISSUE THREAD

**Issue by simahao** [bash-lsp/bash-language-server#241]
vscode:1.45.1
bash-language-server:1.10.1

This plugin  always crash when move mouse or edit bash file. Log like below:

c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\web-tree-sitter\tree-sitter.js:1
var Module=void 0!==Module?Module:{};!function(e,t){"function"==typeof define&&define.amd?define([],t):"object"==typeof exports?module.exports=t():window.TreeSitter=t()}(0,function(){var e,t={};for(e in Module)Module.hasOwnProperty(e)&&(t[e]=Module[e]);var r,n,o=[],s=function(e,t){throw t},_=!1,a=!1;_="object"==typeof window,a="function"==typeof importScripts,r="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node,n=!_&&!r&&!a;var u,i,l,d,c="";r?(c=a?require("path").dirname(c)+"/":__dirname+"/",u=function(e,t){return l||(l=require("fs")),d||(d=require("path")),e=d.normalize(e),l.readFileSync(e,t?null:"utf8")},i=function(e){var t=u(e,!0);return t.buffer||(t=new Uint8Array(t)),q(t.buffer),t},process.argv.length>1&&process.argv[1].replace(/\\/g,"/"),o=process.argv.slice(2),"undefined"!=typeof module&&(module.exports=Module),process.on("uncaug

Error: spawn bash ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
Emitted 'error' event at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:270:12)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:77:11) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn bash',
  path: 'bash',
  spawnargs: [ '-c', 'help function | col -bx' ]
}
[Info  - 下午4:35:41] Connection to server got closed. Server will restart.
[Error - 下午4:35:41] Request textDocument/hover failed.
Error: Connection got disposed.
	at Object.dispose (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\main.js:904:25)
	at Object.dispose (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-languageclient\lib\client.js:74:35)
	at LanguageClient.handleConnectionClosed (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-languageclient\lib\client.js:2309:42)
	at LanguageClient.handleConnectionClosed (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-languageclient\lib\main.js:155:15)
	at closeHandler (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-languageclient\lib\client.js:2296:18)
	at CallbackList.invoke (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\events.js:121:36)
	at closeHandler (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\main.js:240:26)
	at CallbackList.invoke (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\events.js:121:36)
	at IPCMessageReader.fireClose (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at ChildProcess.<anonymous> (c:\Users\Administrator\.vscode\extensions\mads-hartmann.bash-ide-vscode-1.10.1\node_modules\vscode-jsonrpc\lib\messageReader.js:213:45)
	at ChildProcess.emit (events.js:208:15)
	at maybeClose (internal/child_process.js:1021:16)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)

**Comment by nikita-skobov** [comment]
The error message you provided has:

```
errno: 'ENOENT',
code: 'ENOENT',
syscall: 'spawn bash',
path: 'bash',
spawnargs: [ '-c', 'help function | col -bx' ]
}
```

which suggests that it is trying to spawn bash via nodejs `child_process`. What happens if you try to run `bash --version` from your command line? Are you on windows? If so, i don't think bash is installed by default.

Also, I'm not that well versed in this project, but I'm not sure why there is a child process call to spawn bash. Does this happen when you try to open the integrated terminal in vscode, or does this error simply happen when you open a bash file?

**Comment by nikita-skobov** [comment]
UPDATE: ok turns out there was an issue with the `execShellScript` function that wasn't properly capturing errors.

I made a pull request that fixes this: #242 

If/when this gets merged to master, and then a new version of the extension get released, it should fix the problem you are having. 

However, this code is used to find man pages for common programs/commands, so while this fix will prevent crashing, it won't show you the man pages like it should if you don't have bash installed.

As a temporary fix, I would try to make sure bash is installed and available from your path.

**Comment by simahao** [comment]
Thanks for your reply. You are right, I am on windows10, andI use git bash to simulate bash, is it ok?
![image](https://user-images.githubusercontent.com/4537287/82639794-068b8880-9c3c-11ea-9bff-8cbce5498228.png)
Does this happen when you try to open the integrated terminal in vscode?
=>No, it does not happen when I open integreated termianl(git bash)
or does this error simply happen when you open a bash file?
=> Yes, it always happens when I edit bash file or mouse move on the code.

However, this code is used to find man pages for common programs/commands, so while this fix will prevent crashing, it won't show you the man pages like it should if you don't have bash installed.
=>I think it is ok as long as plugin don't crash.

**Comment by nikita-skobov** [comment]
Ok, I see. The pull request will fix the issue if/when it gets merged, but for now, one thing you could try is to make vscode properly detect the path to bash. Bash works through git bash, but I think the issue is that git bash has a different path than other programs use. in other words, try to run `bash --version` on regular windows CMD. Im willing to bet that will give you an error. I don't use windows, so sorry that I cannot give you specific instructions, but I would try to see if you can create some sort of alias for the `bash` keyword to run the bash executable installed by git bash. The issue you described above is caused by your system not being able to run an executable called `bash`, so as long as it is able to find something called `bash` it should work.

**Comment by skovhus** [comment]
Released as bash-language-server@1.16.1 and vs code extension version 1.10.2.

@simahao let me know if it works for you. :)

**Comment by simahao** [comment]
@skovhus Thanks for upgrading the plugin. Now it is ok,very happy, ^_^.
In addition, you said that the plugin should show help information as long as it could find program called bash, I added PATH in windows, so regular CMD could find bash.According to vscode log, I found something like below:
getShellDocumentation failed for "printf" { error: 'Failed to execute help printf | col -bx' }
getShellDocumentation failed for "printf" { error: 'Failed to execute man printf | col -bx' }
I guessed the plugin want to run the "help printf/man printf", because git bash has different format of help command, so i think it may be ok if  add command like "printf --help". BTW, col command is also not supported under git bash, because git bash run under windows, so i think it may be ok even if don't run col command, ^_^.
